### PR TITLE
adding Forte

### PR DESCRIPTION
**Pull request by loriab**
Forte is a research code for computing properties of molecules; it's a plugin to another quantum chemistry program, psi4. Forte is mostly C++ that uses pybind11 to form a Python module. Please see bullet below regarding vendoring.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
  - [x] This does use vendored fmt and nlohmann_json, and the buildsystem and include statements aren't set up to detect external installations instead. As both are header-only, I've added a third-party-license file in hopes that it can be accommodated as-is under (6.b) of https://conda-forge.org/docs/maintainer/guidelines/#generalities . If not, it's fine, and I'll alter the buildsystem. From the developer I understand that fmt will be replaced by `std::fmt` soon anyways.
- [ ] ~If static libraries are linked in, the license of the static library is packaged.~
  - [x] no static libs 
- [ ] ~Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).~
  - [x] no static libs 
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by fevangelista** [comment]
I'm willing to be a maintainer of this package.

**Comment by loriab** [comment]
@conda-forge/help-c-cpp, ready for review!

File: recipes/forte/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+{% set version = "0.3.0" %}
+{% set sha256 = "12b3f8ae7d717721c5109b4c2321ce107fa865f5c509854677c546559b3ab5de" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://github.com/evangelistalab/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (xhochy) [inline recipes/forte/meta.yaml:11]: Please put the sha directly here to minimize the jinja templating.

File: recipes/forte/build.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+if [ "$(uname)" == "Darwin" ]; then
+    CXXFLAGS="${CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"
+
+    # c-f-provided CMAKE_ARGS handles CMAKE_OSX_DEPLOYMENT_TARGET, CMAKE_OSX_SYSROOT
+fi
+ARCH_ARGS=""
+
+${BUILD_PREFIX}/bin/cmake ${CMAKE_ARGS} ${ARCH_ARGS} \
```
Comment (xhochy) [inline recipes/forte/build.sh:8]:
```suggestion
cmake ${CMAKE_ARGS} \
```

File: recipes/forte/build.sh [hunk 3]
```diff
@@ -0,0 +1,30 @@
+if [ "$(uname)" == "Darwin" ]; then
```
Comment (xhochy) [inline recipes/forte/build.sh:1]:
```suggestion
if [[ "${target_platform}" == osx-* ]]; then
```

File: recipes/forte/build.sh [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    CXXFLAGS="${CXXFLAGS} -D_LIBCPP_DISABLE_AVAILABILITY"
+
+    # c-f-provided CMAKE_ARGS handles CMAKE_OSX_DEPLOYMENT_TARGET, CMAKE_OSX_SYSROOT
+fi
+ARCH_ARGS=""
```
Comment (xhochy) [inline recipes/forte/build.sh:6]:
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by loriab** [comment]
Thanks for the review, @xhochy. I think I've addressed the points.

**Review by xhochy — APPROVED** [review]

**Comment by loriab** [comment]
Thanks for the merge!

**Pull request merged**